src/DatePicker/DatePicker.js

```diff
@@ -28,6 +29,63 @@ class DatePicker extends Component {
         this.popoverRef = React.createRef();
     }
 
+    /**
+     * Function tries to format any date string into the format specified by
+     * props.dateFormat (or ISO_DATE_FORMAT as fallback).
+     * It will try to use multiple similar formats, defined in saneDateLookup, to
+     * create a Moment.js date object from given dateString.
+     * It will return an empty string if Moment.js is unable to create a valid
+     * date object from given string (for e.g. text only strings).
+     * @param  {String|Object} date to format.
+     * @returns {String} formatted date string.
+     */
+    getFormattedDateStr = (date) => {
+        if (date) {
+            const momentDateObj = this.getMomentDateObj(date);
+            if (momentDateObj && momentDateObj.isValid()) {
+                const { dateFormat } = this.props;
+                return momentDateObj.format(dateFormat ? dateFormat : ISO_DATE_FORMAT);
+            } else {
+                return '';
+            }
+        }
+        return date;
+    }
+    /**
+     * Function to get date range string from collection of 2 date objects.
+     * @param  {Object[]} dates array containing 2 date objects
+     * @returns {String} formatted date range string e.g. "03/16/2020 - 03/19/2020"
+     */
+    getFormattedDateRangeStr = (dates) => {
+        let str = this.getFormattedDateStr(dates[0]);
+        if (!!dates[1]) {
+            str += dateRangeSeparator + this.getFormattedDateStr(dates[1]);
+        }
+        return str;
+    }
+    /**
+     * Function to create a Moment.js date object from given dateString.
+     * It will try to use multiple similar formats, defined in saneDateLookup, to
+     * parse the given dateString and props.locale
+     * @param  {String} dateString to parse.
+     * @returns {Object} Moment.js date object.
+     */
+    getMomentDateObj(dateString) {
+        return moment(dateString, this.getSaneFormats(), this.props.locale, true);
+    }
+    /**
+     * Get the different date formats allowed for user input.
+     * These formats are based on props.dateFormat
+     * If no entry for props.dateFormat found in saneDateLookup,
+     * it returns the confiured format as string.
+     *
+     * @returns {Array} collection of date formats allowed for input.
+     */
+    getSaneFormats = () =>{
```

Comment:
It's used throughout the files but I'm not sure about `sane`. Maybe `valid` would be more clear?

Reply:
We should include one insane dateFormat- `mm-YYYY/dd` 🤣

Reply:
this is my bad - i started calling these "sane" formats starting 2 weeks ago. Not sure where it came from.

Reply:
I mean.. you're not wrong 🤷‍♀